WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerAdapter.java

```diff
@@ -541,16 +541,22 @@ public int compare(SiteRecord site1, SiteRecord site2) {
             }
 
             if (mSites == null || !mSites.isSameList(sites)) {
-                mAllSites = (SiteList) sites.clone();
-                mSites = filteredSitesByTextIfInSearchMode(sites);
+                SiteList allSites = (SiteList) sites.clone();
+                SiteList filteredSites = filteredSitesByTextIfInSearchMode(sites);
+
+                return new SiteList[]{allSites, filteredSites};
             }
 
             return null;
         }
 
         @Override
-        protected void onPostExecute(Void results) {
-            notifyDataSetChanged();
+        protected void onPostExecute(SiteList[] updatedSiteLists) {
+            if (updatedSiteLists != null) {
```

Comment:
tl;dr no problem 😁  but let me explain my thoughts: I think the aim here was to make sure changes to the class member vars and the call to notifyDatasetChanged are together as much as possible, but this code, while better than before, doesn't really guarantee that. I suggested using a synchronized block and making sure the only call to `notifyDatasetChanged` is within such block as that would be a good way (probably the only way) to guarantee that.
OTOH it is unusual to have to do this to circumvent a problem in the adapter design itself. We can go with this and see if the problem doesn't appear again